azbus/tracing.go

```diff
@@ -8,39 +8,44 @@ import (
 
 func (r *Receiver) handleReceivedMessageWithTracingContext(ctx context.Context, message *ReceivedMessage, handler Handler) (Disposition, context.Context, error) {
 	log := r.log.FromContext(ctx)
+	defer log.Close()
 
 	log.Debugf("ContextFromReceivedMessage(): ApplicationProperties %v", message.ApplicationProperties)
 	var opts = []opentracing.StartSpanOption{}
 	carrier := opentracing.TextMapCarrier{}
 	for k, v := range message.ApplicationProperties {
```

Comment:
This seems very verbose and potentially expensive. Is it really tracing all the application properties ?

Reply:
This is something that JH did so I assume it is a requirement. This PR only adds some debug logging

Reply:
I believe that this just extracts all the message Application Properties into a string map.  That map is then passed into the open tracing constructor which extracts any bits it is interested in to use to setup the spans etc.  It will ignore anything it doesn't care about.  So the filtering of the map is does for us, we don't need to pre-filter it.

This does not trace all the application properties I don't think.

Reply:
thanks

Reply:
assed comment to code